From ac3ad6e4dba097416782f97fc013a28e9370e6b2 Mon Sep 17 00:00:00 2001
From: Camelia Groza <camelia.groza@nxp.com>
Date: Tue, 13 Dec 2016 10:00:42 +0200
Subject: [PATCH 587/666] fsl_qbman: ceetm: set the linked lni id for each
 channel

Also remove whitespace and unnecessary casts.

Signed-off-by: Camelia Groza <camelia.groza@nxp.com>
[Original patch taken from SDK-V2.0-1703]
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/staging/fsl_qbman/qman_high.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/fsl_qbman/qman_high.c b/drivers/staging/fsl_qbman/qman_high.c
index f40f361..4d2cc8f 100644
--- a/drivers/staging/fsl_qbman/qman_high.c
+++ b/drivers/staging/fsl_qbman/qman_high.c
@@ -3853,7 +3853,7 @@ int qman_ceetm_lni_release(struct qm_ceetm_lni *lni)
 	config_opts.dcpid = lni->dcp_idx;
 	memset(&config_opts.shaper_config, 0,
 				sizeof(config_opts.shaper_config));
-	return	qman_ceetm_configure_mapping_shaper_tcfc(&config_opts);
+	return qman_ceetm_configure_mapping_shaper_tcfc(&config_opts);
 }
 EXPORT_SYMBOL(qman_ceetm_lni_release);
 
@@ -4242,6 +4242,7 @@ int qman_ceetm_channel_claim(struct qm_ceetm_channel **channel,
 		return -ENOMEM;
 	p->idx = channel_idx;
 	p->dcp_idx = lni->dcp_idx;
+	p->lni_idx = lni->idx;
 	list_add_tail(&p->node, &lni->channels);
 	INIT_LIST_HEAD(&p->class_queues);
 	INIT_LIST_HEAD(&p->ccgs);
@@ -4325,7 +4326,7 @@ int qman_ceetm_channel_enable_shaper(struct qm_ceetm_channel *channel,
 
 	query_opts.cid = cpu_to_be16(CEETM_COMMAND_CHANNEL_MAPPING |
 						channel->idx);
-	query_opts.dcpid = (u8)channel->dcp_idx;
+	query_opts.dcpid = channel->dcp_idx;
 
 	if (qman_ceetm_query_mapping_shaper_tcfc(&query_opts, &query_result)) {
 		pr_err("Can't query channel mapping\n");
-- 
2.7.4

